cli/command/image/build_buildkit.go

```diff
@@ -43,6 +46,13 @@ func runBuildBuildKit(dockerCli command.Cli, options buildOptions) error {
 		return errors.Errorf("buildkit not supported by daemon")
 	}
 
+	if options.imageIDFile != "" {
```

Comment:
I'm sure I'm missing some context for where this gets run, but this seems odd.
What if the user already has this file from a previous build and doesn't want it removed?

Reply:
@cpuguy83 this seems to be the behavior for old builder as well. The idea is to say, if we can't remove the file, we can't write to it, so fail early instead of running the whole successful build and then fail at writing the image id file.